app/models/reports/repository.rb

```diff
@@ -155,17 +163,46 @@ def initialize(regions, periods:)
     end
 
     memoize def controlled
+      return controlled_v2 if reporting_schema_v2?
+      science("controlled") do |e|
+        e.use { controlled_v1 }
+        e.try { controlled_v2 }
+        e.raise_on_mismatches = true
+      end
+    end
+
+    def controlled_v1
```

Comment:
I would guess we can make these v1/v2 methods private. If you want to keep them close to the methods that call them, you can do the inline `private def`.